### PR TITLE
Lock masquerade object during json.Marshal

### PR DESCRIPTION
**Pull request by atavism**
Resolves https://github.com/getlantern/engineering/issues/1546

### CODE REVIEW

**Review by WendelHime — APPROVED** [review]
Tested locally three times and no data race detected!

**Comment by atavism** [comment]
Thanks for the review, @WendelHime!!

File: masquerade.go [hunk 1]
```diff
@@ -48,6 +49,15 @@ type masquerade struct {
 	mx         sync.RWMutex
 }
 
+// MarshalJSON marshals masquerade into json
+func (m *masquerade) MarshalJSON() ([]byte, error) {
+	m.mx.RLock()
+	defer m.mx.RUnlock()
+	// Type alias for masquerade so that we don't infinitely recurse when marshaling the struct
+	type alias masquerade
```
Comment (hwh33) [inline masquerade.go:57]: Nifty!

**Review by hwh33 — APPROVED** [review]
LGTM

**Comment by atavism** [comment]
Thanks, @hwh33!

**Pull request merged**